tests/System.Text.Primitives.Encoding.Performance.Tests/EncodingPerfComparisonTests.cs

```diff
@@ -0,0 +1,68 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Text.Utf8;
+using Microsoft.Xunit.Performance;
+
+
+namespace System.Text.Primitives.Encoding.Performance.Tests
+{
+    public class EncodingPerfComparisonTests
+    {
+        [Benchmark]
+        public void EncodingPerformanceTestUsingCoreCLR()
+        {
+            string unicodeString = GenerateString(1000);
+            ReadOnlySpan<char> characters = unicodeString.Slice();
+
+            Text.Encoding utf8 = Text.Encoding.UTF8;
+
+            int utf8Length = utf8.GetByteCount(unicodeString);
+            byte[] utf8Buffer = new byte[utf8Length];
+            Span<byte> span;
+
+            char[] charArray = characters.ToArray();
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                    utf8.GetBytes(charArray, 0, characters.Length, utf8Buffer, 0);
+                span = new Span<byte>(utf8Buffer);
+            }
+        }
+
+        [Benchmark]
+        public void EncodingPerformanceTestUsingCorefxlab()
+        {
+            string unicodeString = GenerateString(1000);
+            ReadOnlySpan<char> characters = unicodeString.Slice();
+
+            int encodedBytes;
+
+            int utf8Length = Utf8Encoder.ComputeEncodedBytes(characters);
+            byte[] utf8Buffer = new byte[utf8Length];
+            Span<byte> span = new Span<byte>(utf8Buffer);
+
+            foreach (var iteration in Benchmark.Iterations)
+            {
+                using (iteration.StartMeasurement())
+                    if (!Utf8Encoder.TryEncode(characters, span, out encodedBytes))
+                    {
+                        throw new Exception(); // this should not happen
+                    }
+            }
+        }
+
+        public string GenerateString(int charLength)
+        {
+            Random rand = new Random();
+            var plainText = new StringBuilder();
+            for (int j = 0; j < charLength; j++)
+            {
+                plainText.Append((char)rand.Next(char.MinValue, char.MaxValue));
```

Comment:
String composed of completely random Unicode characters like this do not exist in practice. The strings you generate this way will tend to also be invalid because of they will contain invalid surrogate combinations - I assume that's why you had to do the roundtrip using Encoding.Unicode to get rid of the invalid surrogates.

Instead, you should test:

- ASCII strings (ie 1 byte UTF8 encoding)
- Cyrilic strings (ie 2 byte UTF8 encoding) 
- Chinese string (ie 3 byte UTF8 encoding)
- Combinations of the above